pyvista/core/dataobject.py

```diff
@@ -189,39 +189,172 @@ def copy(self, deep=True):
         Parameters
         ----------
         deep : bool, optional
-            When True makes a full copy of the object.
+            When ``True`` makes a full copy of the object.  When
+            ``False``, performs a shallow copy where the points, cell,
+            and data arrays are pointers to the original object.
 
         Returns
         -------
-        newobject : same as input
-           Deep or shallow copy of the input.
+        :class:`pyvista.DataSet`
+            Deep or shallow copy of the input.
+
+        Examples
+        --------
+        Create and make a deep copy of a PolyData object.
+
+        >>> import pyvista
+        >>> mesh_a = pyvista.Sphere()
+        >>> mesh_b = mesh_a.copy()
+        >>> mesh_a == mesh_b
+        True
 
         """
         thistype = type(self)
         newobject = thistype()
+
+        # this must be called to ensure scalars active are "active"
+        # before performing a deep copy
+        self.active_scalars_info
```

Comment:
There's also `active_vectors_info`, `active_tensors_info` at least on `PolyData`. Are these not necessary to be accessed before copying?

Reply:
Will check.

Reply:
Problem is that there's a side effect with running `active_scalars_info`.  Investigating...

Reply:
This was quite a bug.  Turns out returning `active_scalars_info` would provide the wrong active scalars in some cases, reset them in others.  It's been cleaned up in c27fbb269a3e3ca22da4591808cec75aa0e07ce6 and 90efb773c7069d636b78e05bd6ebb260a31877fb.